nixos/modules/services/mail/mailman.nix

```diff
@@ -0,0 +1,120 @@
+{ config, pkgs, lib, ... }:          # mailman.nix
+
+with lib;
+
+let
+
+  cfg = config.services.mailman;
+
+  pythonEnv = pkgs.python3.withPackages (ps: [ps.mailman]);
+
+  mailmanExe = with pkgs; stdenv.mkDerivation {
+    name = "mailman-" + python3Packages.mailman.version;
+    unpackPhase = ":";
+    installPhase = ''
+      mkdir -p $out/bin
+      sed >"$out/bin/mailman" <"${pythonEnv}/bin/mailman" \
+        -e "2 iexport MAILMAN_CONFIG_FILE=/etc/mailman.cfg"
+      chmod +x $out/bin/mailman
+    '';
+  };
+
+  mailmanCfg = ''
+    [mailman]
+    site_owner: ${cfg.siteOwner}
+    layout: fhs
+
+    [paths.fhs]
+    bin_dir: ${pkgs.python3Packages.mailman}/bin
+    var_dir: /var/lib/mailman
+    queue_dir: $var_dir/queue
+    log_dir: $var_dir/log
+    lock_dir: $var_dir/lock
+    etc_dir: /etc
+    ext_dir: $etc_dir/mailman.d
+    pid_file: /run/mailman/master.pid
+  '';
+
+in {
+
+  ###### interface
+
+  options = {
+
+    services.mailman = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable Mailman on this host. Requires an active Postfix installation.";
+      };
+
+      siteOwner = mkOption {
+        type = types.str;
+        default = "postmaster";
+        description = ''
+          Certain messages that must be delivered to a human, but which can't
+          be delivered to a list owner (e.g. a bounce from a list owner), will
+          be sent to this address. It should point to a human.
+        '';
+      };
+
+
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    assertions = [
+      { assertion = cfg.enable -> config.services.postfix.enable;
+        message = "Mailman requires Postfix";
+      }
+      { assertion = config.services.postfix.recipientDelimiter == "+";
+        message = "Postfix's recipientDelimiter must be set to '+'.";
+      }
+    ];
+
+    users.users = singleton {
+      name = "mailman";
+      group = "mailman";
+      uid = config.ids.uids.mailman;
```

Comment:
Not sure why you pushed these id reserves [to master](https://github.com/NixOS/nixpkgs/commit/95c021393f3)? PR's are there to get a second pair of eyes to look at changes, I gave you a review in [the previous PR](https://github.com/NixOS/nixpkgs/pull/67461#discussion_r318253808), but arguing that a change is fine because you already pushed it to master isn't alright.

Reply:
> Not sure why you pushed these id reserves [to master](https://github.com/NixOS/nixpkgs/commit/95c021393f3)?

Why wouldn't I? I've been pushing trivial changes to `master` for the last 11 years. 

> I gave you a review in [the previous PR](https://github.com/NixOS/nixpkgs/pull/67461#discussion_r318253808), but arguing that a change is fine because you already pushed it to master isn't alright.

I never argued that this change is fine. I argued that since the change was already in `master`, removing those ids again is a separate topic.

Reply:
You've both been long-time contributors with a good feel for the landscape of the NixOS ecosystem, and you both do a lot of really good work. I know not everybody has decided to move to using PRs but for what it is worth I think overall the community has been moving to a mentality that directly pushing patches should ultimately be deprecated.

Reply:
6b87772ca4a0b7d8448365b4f1c1a08a904fb3e8